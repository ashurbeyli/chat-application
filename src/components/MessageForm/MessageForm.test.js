import React from 'react';
import { shallow } from 'enzyme';

import { MessageForm } from './index';

const mockMessage = 'message';

describe('MessageForm', () => {
  const wrapper = shallow(<MessageForm />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render form', () => {
    expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper.find('input').exists()).toBe(true);
    expect(wrapper.find('button').exists()).toBe(true);
  });

  describe('should update message', () => {

    beforeEach(() => {
      wrapper.find('input').simulate('change', { target: { name: 'message', value: mockMessage} });
    });

    it('should update message in state', () => {
      expect(wrapper.state().message).toEqual(mockMessage);
    });
  });

});