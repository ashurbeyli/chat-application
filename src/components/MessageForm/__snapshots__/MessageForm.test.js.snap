// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="send-form"
        method="POST"
        onSubmit={[Function]}
>
        <input
                autoComplete="off"
                className="send-form__textInput"
                name="message"
                onChange={[Function]}
                placeholder="Message"
                type="text"
                value=""
        />
        <button
                className="send-form__sendButton"
        >
                Send
        </button>
</form>,
      "className": "block-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            autoComplete="off"
            className="send-form__textInput"
            name="message"
            onChange={[Function]}
            placeholder="Message"
            type="text"
            value=""
/>,
          <button
            className="send-form__sendButton"
>
            Send
</button>,
        ],
        "className": "send-form",
        "method": "POST",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "off",
            "className": "send-form__textInput",
            "name": "message",
            "onChange": [Function],
            "placeholder": "Message",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Send",
            "className": "send-form__sendButton",
          },
          "ref": null,
          "rendered": "Send",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="send-form"
          method="POST"
          onSubmit={[Function]}
>
          <input
                    autoComplete="off"
                    className="send-form__textInput"
                    name="message"
                    onChange={[Function]}
                    placeholder="Message"
                    type="text"
                    value=""
          />
          <button
                    className="send-form__sendButton"
          >
                    Send
          </button>
</form>,
        "className": "block-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoComplete="off"
              className="send-form__textInput"
              name="message"
              onChange={[Function]}
              placeholder="Message"
              type="text"
              value=""
/>,
            <button
              className="send-form__sendButton"
>
              Send
</button>,
          ],
          "className": "send-form",
          "method": "POST",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "className": "send-form__textInput",
              "name": "message",
              "onChange": [Function],
              "placeholder": "Message",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send",
              "className": "send-form__sendButton",
            },
            "ref": null,
            "rendered": "Send",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
